control/HTTP.php

```diff
@@ -372,6 +371,10 @@ public static function add_cache_headers($body = null) {
 		if(headers_sent() && !$body) {
 			return;
 		}
+		// Skip already assigned cache-control headers
+		if ($body && $body->getHeader('Cache-Control')) {
```

Comment:
That seems a bit dangerous in terms of footguns, e.g. devs might miss the fact that we have a highlevel API and add go the obvious route instead, using `SS_HTTPResponse->addHeader()`. In particular, it doesn't protect any devs who used this lowlevel API in existing projects by default. I think `HTTPCacheControl` needs to override direct setting of `Cache-Control` headers. We should also note this in `addHeaders()` PHPDoc.

Reply:
Discussed this with Sam and Damian, we'll retain the warning for custom `addHeader()` uses (which you should be warned about), and deprecate the entire controllerpolicy module (to avoid situations where the warning doesn't apply).